vulnerabilities/importers/apache_kafka.py

```diff
@@ -7,120 +7,177 @@
-                    [
-                        MavenVersion(version) in version_range
-                        for version_range in fixed_version_ranges
-                    ]
-                )
-            ]
 
-            affected_packages = [
-                PackageURL(type="apache", name="kafka", version=version)
-                for version in self.version_api.get("apache/kafka").valid_versions
-                if any(
-                    [
-                        MavenVersion(version) in version_range
-                        for version_range in affected_version_ranges
-                    ]
-                )
+            affected_versions_string = affected_versions_row.find_all("td")[1].text
+            fixed_versions_string = fixed_versions_row.find_all("td")[1].text
+
+            # Remove leading white space after initial comma
+            affected_versions_string_split_SPLIT = [
+                substring.strip()
+                for substring in affected_versions_string.split(",")
+                if not substring.isspace()
+            ]
+            fixed_versions_string_split_SPLIT = [
+                substring.strip()
+                for substring in fixed_versions_string.split(",")
+                if not substring.isspace()
             ]
 
-            advisories.append(
-                AdvisoryData(
-                    vulnerability_id=cve_id,
-                    summary=cve_description_paragraph.text,
-                    affected_packages=nearest_patched_package(affected_packages, fixed_packages),
-                    references=[
-                        Reference(url=ASF_PAGE_URL),
-                        Reference(
-                            url=f"https://cve.mitre.org/cgi-bin/cvename.cgi?name={cve_id}",
-                            reference_id=cve_id,
-                        ),
+            # This throws a KeyError if the opening h2 tag `id` data changes or is not in the
+            # hard-coded affected_version_range_mapping dictionary.
+            if affected_version_range_mapping[cve_id]["action"] == "include":
+
+                # These 2 variables (not used elsewhere) trigger the KeyError for changed/missing data.
+                check_affected_versions_key = affected_version_range_mapping[cve_id][
+                    affected_versions_string
+                ]
+                check_fixed_versions_key = affected_version_range_mapping[cve_id][
+                    fixed_versions_string
+                ]
+
+                references = [
+                    Reference(url=self.ASF_PAGE_URL),
+                    Reference(
+                        url=f"https://cve.mitre.org/cgi-bin/cvename.cgi?name={cve_id}",
+                        reference_id=cve_id,
+                    ),
+                ]
+
+                affected_packages = []
+                affected_package = AffectedPackage(
+                    package=PackageURL(
+                        name="apache_kafka",
```

Comment:
name should be kafka and type should be apache IMO

Reply:
Change made -- thank you. 👍